src/main/kotlin/apply/security/BearerAuthInterceptor.kt

```diff
@@ -0,0 +1,42 @@
+package apply.security
+
+import org.springframework.stereotype.Component
+import org.springframework.web.method.HandlerMethod
+import org.springframework.web.servlet.HandlerInterceptor
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+private const val AUTHORIZATION_HEADER = "Authorization"
+private const val BEARER = "Bearer"
+private const val TOKEN_TYPE_INDEX = 0
+private const val TOKEN_INDEX = 1
+private const val VALID_CHUNK_SIZE = 2
+
+@Component
+class BearerAuthInterceptor(private val jwtTokenProvider: JwtTokenProvider) : HandlerInterceptor {
+
+    override fun preHandle(request: HttpServletRequest, response: HttpServletResponse, handler: Any): Boolean {
+        val handlerMethod = handler as? HandlerMethod ?: return true
+
+        if (handlerMethod.methodParameters.none { it.hasParameterAnnotation(LoginApplicant::class.java) }) {
+            return true
+        }
+
+        val token = extractBearerToken(request)
+        val applicantEmail = jwtTokenProvider.getPayload(token)
```

Comment:
[jwt](https://velopert.com/2389) 글에 의하면 jwt token의 payload에는 토큰에 담긴 모든 정보가 들어있습니다!
token claim에 해당하는 subject의 값을 가져오는 것이니 getSubject라는 이름이 더 적절해보여요 :)
```suggestion
        val applicantEmail = jwtTokenProvider.getSubject(token)
```